src/core/derived_state.py

```diff
@@ -0,0 +1,57 @@
+from dataclasses import dataclass
+import numpy as np
+from typing import Dict
+
+@dataclass
+class DerivedState:
+    """
+    Container class for derived state variables needed for state determination.
+    TODO: Concretely document these somewhere.
+    """
+
+    # inertia matrix components (kg * m^2). Structure is
+    # [[Ixx, Ixy, Ixz], 
+    #  [Iyx, Iyy, Iyz], 
+    #  [Izx, Izy, Izz]].
+    Ixx: float = 0.0
+    Ixy: float = 0.0
+    Ixz: float = 0.0
+    Iyx: float = 0.0
+    Iyy: float = 0.0
+    Iyz: float = 0.0
+    Izx: float = 0.0
+    Izy: float = 0.0
+    Izz: float = 0.0
+
+    # Kane damping constant
+    kane_c: float = 0.0
```

Comment:
Is this a constant parameter or a truly state-dependent number?

Reply:
This is determined by the tank fill fraction, so it is state-dependent

Reply:
Ah, right!